chore: Increase `agenda` time in Omnichannel EE jobs to 1 minute

KevLehman opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

<!-- Your Pull Request name should start with one of the following tags
  feat: Adding a new feature
  refactor: A code change that doesn't change behavior (it doesn't add anything and doesn't fix anything)
  fix: For bug fixes that affect the end-user
  chore: For small tasks
  docs: For documentation
  ci: For updating CI configuration
  test: For adding tests
  i18n: For updating any translations
  regression: Issues created/reported/fixed during the development phase. kind of problem that never existed in production and that we don't need to list in a changelog for the end user
-->

<!-- Checklist!!! If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. 
  - I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  - I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - Lint and unit tests pass locally with my changes
  - I have added tests that prove my fix is effective or that my feature works (if applicable)
  - I have added necessary documentation (if applicable)
  - Any dependent changes have been merged and published in downstream modules
-->

## Proposed changes (including videos or screenshots)
<!--
  Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
  If it fixes a bug or resolves a feature request, be sure to link to that issue below.
  This description won't be displayed to our end users in the release notes, so feel free to add as much technical context as needed.
  If the changes introduced in this pull request must be presented in the release notes, make sure to add a changeset file. Check our guidelines for adding a changeset to your pull request: https://developer.rocket.chat/contribute-to-rocket.chat/modes-of-contribution/participate-in-rocket.chat-development/development-workflow#4.-adding-changeset-to-your-pull-request 
-->

## Issue(s)
<!-- Link the issues being closed by or related to this PR. For example, you can use #594 if this PR closes issue number 594 -->

## Steps to test or reproduce
<!-- Mention how you would reproduce the bug if not mentioned on the issue page already. Also mention which screens are going to have the changes if applicable -->

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->
This is to reduce the frequency of these actions. These 3 schedulers were left behind when we moved away from synced-cron to our own cron package. Other task will be created to move these schedulers to use it in the future.

MarcosSpessatto commented:
Since we are skipping the test covering this, we will create unit tests to cover the whole class in a subsequent PR.

ggazzo reviewed (changes requested):
There is no way this could be a patch release, There is no problem being solved here, at most we are masking something worse. I don't believe we should release this as a fix.

Furthermore, the title points out to something different.

Skipping a test is not ideal, you could make this modifiable via an envvar

KevLehman commented:
Problem being solved is that the default timing for OM jobs is 5 secs, while the default timer for any other Agenda job (via @rocket.chat/cron) is 1 minute. These items were left when refactoring from old jobs to /cron package.

While I'd agree there's no "problem" being solved, the sole fact these jobs are going to the db every 5s (which also caused problems in the past when we first migrated to /cron) is problematic by itself.

What title would you suggest? Title says we're increasing agenda time, maybe that part is not clear but what we're actually doing is increasing the agenda processing time.

About the test, yeah, skipping is not ideal, but since this is targetted for an older version, we decided to not to add more stuff in here and just skip for now. Idea is to make these jobs configurable in the future, just not now.

wdyt?

ggazzo commented:
I didn't understand the part about targeting an old version, as it is pointing to develop

about the title, `chore`, generally does not indicate fix, but some internal modification. Doesn't sound like a real fix

maybe something like fix: Omnichannel Jobs running every 5 seconds

KevLehman commented:
All right, updated PR title. I actually didn't notice I put this one as a chore. Fixed now.

The part about and old version is from a customer that's getting bombed by this. When we first moved to `cron` package, we had the same issue of DB getting bombed by agenda and so the fix was to update it to run every min. These 3 jobs were left behind on that fix (probably since the fix was spot on Open and we don't have these running there) and so the same issue persisted for these, and now a customer is having problems because of it (not all the problems are tied to these, but fixing them would alliviate a bit)

ggazzo commented:
/patch
